api/nodes/views.py

```diff
@@ -1394,7 +1397,20 @@ def get_object(self):
             self.kwargs[node_link_lookup_url_kwarg],
             'node link'
         )
-        # May raise a permission denied
+        node = get_object_or_error(
```

Comment:
Very nearly. https://github.com/CenterForOpenScience/osf.io/pull/5897#discussion_r71893066 suggested "Most or all of the stuff you did below should be done by the permissions class. Take a look at how ContributorOrPublicForPointers works; it loads the node object and does checks on that." 

Rather than keeping all of this in the get object method, it should be part of the permission classes. The example of how to do that is in ContributorOrPublicForPointers.